django/forms/fields.py

```diff
@@ -873,8 +872,8 @@ def has_changed(self, initial, data):
             data = []
         if len(initial) != len(data):
             return True
-        initial_set = set(force_text(value) for value in initial)
-        data_set = set(force_text(value) for value in data)
+        initial_set = set(str(value) for value in initial)
```

Comment:
This could use set comprehension also. There was another commit that converted most places but perhaps this slipped through.